splunk_connect_for_snmp_poller/manager/poller.py

```diff
@@ -69,13 +71,39 @@ def should_process_current_line(self, host, version, community, profile, frequen
             host, version, community, profile, frequency
         )
 
-    def check_inventory(self):
+    def check_server_config(self):
+        server_config_file = self._args.config
+
+        is_config_changed = False
+        # detect the config changes
+        if (
+            os.stat(server_config_file, follow_symlinks=True).st_mtime
+            > self._config_mod_time
+        ):
+            logger.info("[-] Change in scheduler config detected, reloading")
+            self._config_mod_time = os.stat(
+                server_config_file, follow_symlinks=True
+            ).st_mtime
+            is_config_changed = True
```

Comment:
This code is a copy&paste. Can you please define in another file (call it file_utilities, or whatever you like) a function called:
def file_was_modified(file_path):
- put in here all the os.stat() logic
- this function should return True/False

In this way, the poller could be rewritten as:
def check_server_config(self):
    self.check_inventory(parse_config_file(self._args.config), file_was_modified(self._args.config))

def check_inventory(self, server_config, is_config_changed):
    if file_was_modified(self._args.inventory) or is_config_changed: 
        "open self._args.inventory and read it..."

Reply:
👍 Thanks for your valuable comments. :) 
Defined the file_was_modified inside file utilities.py